### PR TITLE
[SHARE-1016][Improvement] Upgrade ember-cli

### PR DESCRIPTION
**Pull request by laurenbarker**
## TODO
- [x] Reset page query param when other params change
- [x] Get rid of the rest of the eslint warnings
- [x] Upgrade to 2.18
- [x] Include headless chrome too
- [x] Fix building facets

## Purpose
Upgrade ember-cli from 2.5 to 2.18.

## Changes
* Includes removing more dead code
* Includes changes from #195 and #196 
* Removed graph adapter and just used ajax
* Refactored pages to use ember-concurrency and ember-parachute

### CODE REVIEW

File: app/components/section-related-works/component.js [hunk 1]
```diff
@@ -38,16 +52,17 @@ export default Ember.Component.extend({
                                 ${PAGE_FRAGMENT_MAP.AbstractAgent.relatedWorks(offset)}
                             }
                         }
-                    }`
-                }
-            }).then(data => {
-                if (data.errors) {throw Error(data.errors[0].message);}
```
Comment (laurenbarker) [inline app/components/section-related-works/component.js]: Add this back. Remove try catch.

File: app/controllers/discover.js [hunk 2]
```diff
@@ -360,39 +273,23 @@ export default ApplicationController.extend({
...
-        typing(val, event) {
-            // Ignore all keycodes that do not result in the value changing
-            // 8 == Backspace, 32 == Space
-            if (event.keyCode < 49 && !(event.keyCode === 8 || event.keyCode === 32)) {
-                return;
-            }
-            this.search();
-        },
-
         search() {
```
Comment (laurenbarker) [inline app/controllers/discover.js]: Refresh search

File: app/controllers/discover.js [hunk 3]
```diff
@@ -428,17 +323,186 @@ export default ApplicationController.extend({
...
+    setup({ queryParams }) {
+        this.get('fetchData').perform(queryParams);
+    },
+
+    queryParamsDidChange({ shouldRefresh, queryParams, changed }) {
+        if (shouldRefresh) {
+            this.get('fetchData').perform(queryParams);
+        }
+
+        if (queryParams.page !== 1 && !changed.page) {
```
Comment (laurenbarker) [inline app/controllers/discover.js]: Move this above

File: app/mixins/detail.js [hunk 4]
```diff
@@ -1,32 +1,37 @@
...
     // Links that people could actual click on and not get XML/JSON/etc
-    links: Ember.computed('model.identifiers', function() {
+    links: computed('model.identifiers', function() {
         return this.get('model.identifiers').filter(identifier =>
-            Object.keys(VISITABLE).reduce((acc, k) => acc || VISITABLE[k].test(identifier[k]), false) &&
-            !Object.keys(AVOID).reduce((acc, k) => acc || AVOID[k].test(identifier[k]), false)
+            Object.keys(VISITABLE).reduce((acc, k) => acc ||
+            VISITABLE[k].test(identifier[k]), false) &&
+            !Object.keys(AVOID).reduce((acc, k) => acc ||
+            AVOID[k].test(identifier[k]), false),
```
Comment (laurenbarker) [inline app/mixins/detail.js]: Can this be fit on 3 lines?

File: app/routes/detail.js [hunk 5]
```diff
@@ -1,38 +1,61 @@
...
+import ENV from '../config/environment';
+
+
+export default Route.extend(RouteHistoryMixin, {
+    session: service(),
+
     model(params) {
-        let adapter = this.store.adapterFor('graph');
-        return adapter.ajax('/api/v2/graph/', 'POST', {
+        return $.ajax({
```
Comment (laurenbarker) [inline app/routes/detail.js]: Add comment with the graphQL ticket here

File: app/templates/discover.hbs [hunk 6]
```diff
@@ -3,7 +3,7 @@
     <br>
     <div class="row share-search">
         <div class="col-xs-8 col-xs-offset-2">
-            {{input class="form-control" value=q key-up='typing' placeholder=placeholder}}
+            {{input class="form-control" value=q placeholder=placeholder}}
```
Comment (laurenbarker) [inline app/templates/discover.hbs]: Use replace for q

Comment (laurenbarker) [reply app/templates/discover.hbs]: `replace: true` on q didn't work so I'm going to defer this improvement :/

File: app/components/section-related-works/component.js (hunk 1 above)
Comment (laurenbarker) [reply app/components/section-related-works/component.js]: Done.

File: app/components/search-facet-worktype-button/component.js [hunk 7]
```diff
@@ -1,28 +1,30 @@
...
             this.$().blur();
-            this.sendAction('onClick', this.get('type'));
+            this.onClick(this.get('type'));
         },
 
         toggleBody() {
-            this.sendAction('toggleCollapse');
-        }
-    }
+            this.toggleCollapse();
```
[the remark this answers is not in the document]
Comment (laurenbarker) [reply app/components/search-facet-worktype-button/component.js]: It seems that in [blog posts](https://www.axiomq.com/blog/introduction-to-ember-closure-actions/) about closure actions people tend to just call the action directly on the component (`this.action()`) but in the [eslint rule description](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/closure-actions.md) they get the action first (`get(this, 'action')()`). I'll get the action for consistency.

I don't think the [ember docs](https://www.emberjs.com/api/ember/2.16/classes/Ember.Templates.helpers/methods/action?anchor=action) cover the syntax.

[the remark this answers is not in the document]
Comment (aaxelb) [reply app/components/search-facet-worktype-button/component.js]: The [ember guides](https://guides.emberjs.com/v2.18.0/components/triggering-changes-with-actions/#toc_passing-the-action-to-the-component) do suggest using `get`, probably mostly for consistency.
```js
// call the onConfirm property to invoke the passed in action
this.get('onConfirm')();
```
It does seem cleaner to always do it the same way, so we don't have to think about special cases.

[the remark this answers is not in the document]
Comment (laurenbarker) [reply app/components/search-facet-worktype-button/component.js]: Ah there it is!

File: app/controllers/discover.js (hunk 2 above)
Comment (laurenbarker) [reply app/controllers/discover.js]: Done.

**Comment by aaxelb** [comment]
Tested locally, all the functionality looks good! There are a few weird layout changes, though. Vertical spacing in the navbar looks odd, and there's no longer horizontal margin on either side.

Prod:
screen shot 2018-01-12 at 12 27 56 pm
screen shot 2018-01-12 at 12 32 42 pm

Local:
screen shot 2018-01-12 at 12 27 19 pm
screen shot 2018-01-12 at 12 32 29 pm

**Pull request merged**